src/hooks/useEvaluateFormFieldExpressions.ts

```diff
@@ -125,7 +126,7 @@ export const useEvaluateFormFieldExpressions = (
         }
       });
     });
-    setEvaluatedFormJson(factoryContext.formJson);
+    setEvaluatedFormJson(updateFormSectionReferences(factoryContext.formJson));
```

Comment:
There seems to be a risk with multiple re-renders with this but i can't see a way out of it. Most complex forms will have multiple expressions.

Reply:
The impact on performance isn’t that critical. The hook is called once:

<img width="761" alt="Screenshot 2024-10-28 at 23 52 58" src="https://github.com/user-attachments/assets/f68ac246-5b22-4113-ab11-448c10872ca0">